src/plugins/unified_search/public/saved_query_management/saved_query_management_list.tsx

```diff
@@ -165,58 +283,57 @@ export function SavedQueryManagementList({
   }, []);
 
   const onDelete = useCallback(
-    (savedQueryToDelete: string) => {
+    (savedQueryToDelete: SavedQuery) => {
       const onDeleteSavedQuery = async (savedQueryId: string) => {
-        cancelPendingListingRequest.current();
-        setSavedQueries(
-          savedQueries.filter((currentSavedQuery) => currentSavedQuery.id !== savedQueryId)
+        setTotalQueryCount((currentTotalQueryCount) => max([0, currentTotalQueryCount - 1])!);
```

Comment:
```suggestion
        setTotalQueryCount((currentTotalQueryCount) => Math.max(0, currentTotalQueryCount - 1)!);
```

Reply:
Too much coding is making me forgetful 🙂 The standard lib is always preferable, changed it here: https://github.com/elastic/kibana/pull/170599/commits/032a918e90b01c217f98b299ef18b9c578a184d2.